dashboard/modules/metrics/install_and_start_prometheus.py

```diff
@@ -0,0 +1,192 @@
+import logging
+import os
+import platform
+import subprocess
+import sys
+import tarfile
+from pathlib import Path
+
+import requests
+
+from ray.dashboard.consts import PROMETHEUS_CONFIG_INPUT_PATH
+
+# Configure basic logging
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+FALLBACK_PROMETHEUS_VERSION = "2.48.1"
+DOWNLOAD_BLOCK_SIZE = 8192  # 8 KB
+TEST_MODE_ENV_VAR = "RAY_PROMETHEUS_DOWNLOAD_TEST_MODE"
+
+
+def get_system_info():
+    os_type = platform.system().lower()
+    architecture = platform.machine()
+    if architecture == "x86_64":
+        # In the Prometheus filename, it's called amd64
+        architecture = "amd64"
+    return os_type, architecture
+
+
+def download_file(url, filename):
+    logging.info(f"Downloading {url} to {Path(filename).absolute()}...")
+    try:
+        test_mode = os.environ.get(TEST_MODE_ENV_VAR, False)
+        request_method = requests.head if test_mode else requests.get
+        response = request_method(url, stream=True)
+        response.raise_for_status()
+
+        total_size_in_bytes = int(response.headers.get("content-length", 0))
+        total_size_in_mb = total_size_in_bytes / (1024 * 1024)
+
+        downloaded_size_in_mb = 0
+        block_size = DOWNLOAD_BLOCK_SIZE
+
+        with open(filename, "wb") as file:
+            for chunk in response.iter_content(chunk_size=block_size):
+                file.write(chunk)
+                downloaded_size_in_mb += len(chunk) / (1024 * 1024)
+                print(
+                    f"Downloaded: {downloaded_size_in_mb:.2f} MB / "
+                    f"{total_size_in_mb:.2f} MB",
+                    end="\r",
+                )
+
+        print("\nDownload completed.")
+        return True
+
+    except requests.RequestException as e:
+        logging.error(f"Error downloading file: {e}")
+        return False
+
+
+def install_prometheus(file_path):
+    try:
+        with tarfile.open(file_path) as tar:
+            tar.extractall()
+        logging.info("Prometheus installed successfully.")
+        return True
+    except Exception as e:
+        logging.error(f"Error installing Prometheus: {e}")
+        return False
+
+
+def start_prometheus(prometheus_dir):
+
+    config_file = Path(PROMETHEUS_CONFIG_INPUT_PATH)
+
+    if not config_file.exists():
+        raise FileNotFoundError(f"Prometheus config file not found: {config_file}")
```

Comment:
This configuration only gets created once ray starts. It might make sense to see if we can move the config generation logic to this file instead since this runs before Ray starts.

Reply:
Discussed offline: 
- Currently in Ray, "creating the configuration" just consists of copying a hardcoded config from the site-packages Ray directory into some location in `/tmp/ray`.  In this PR, we pull the config file from the first location, so we don't actually have to run Ray first.
- However, if in the future we make a change to Ray that makes updates to the hardcoded config upon `ray start` before copying it to `/tmp/ray`, then `install_and_start_prometheus.py` won't have the updated config. So we should at the very least add code comments in the install script and in the "Ray creating the config" step to make sure future maintainers take this into account. I'll update the PR with these comments.